packages/app-store/pipedrive-crm/api/add.ts

```diff
@@ -0,0 +1,30 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { createDefaultInstallation } from "@calcom/app-store/_utils/installation";
+import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
+import { HttpError } from "@calcom/lib/http-error";
+
+import appConfig from "../config.json";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "GET") return res.status(405).json({ message: "Method not allowed" });
+  // Check that user is authenticated
+  req.session = await getServerSession({ req, res });
+  const { teamId } = req.query;
+  const userId = req.session?.user.id;
+  if (!userId) {
+    throw new HttpError({ statusCode: 401, message: "You must be logged in to do this" });
+  }
+  await createDefaultInstallation({
+    appType: `${appConfig.slug}_other_calendar`,
+    userId: userId,
+    slug: appConfig.slug,
+    key: {},
+    teamId: Number(teamId),
+  });
+  const tenantId = teamId ? teamId : userId;
+  res.status(200).json({
+    url: `https://oauth.pipedrive.com/oauth/authorize?client_id=${process.env.PIPEDRIVE_CLIENT_ID}&redirect_uri=https://app.revert.dev/oauth-callback/pipedrive&state={%22tenantId%22:%22${tenantId}%22,%22revertPublicToken%22:%22${process.env.REVERT_PUBLIC_TOKEN}%22}`,
```

Comment:
<img width="927" alt="Screenshot 2023-12-19 at 12 40 30 PM" src="https://github.com/calcom/cal.com/assets/1780212/d2115fa2-482f-4d88-8d25-f0fcf697b322">
You should read the pipedrive client_id and secret like this